docs/DeveloperGuide.md

```diff
@@ -1424,6 +1427,182 @@ testers are expected to do more *exploratory* testing.
+   Expected: 1st wedding's date is set as `John's Wedding`, with address `Address1` and date `2024-12-31`. Details of edited wedding are shown.
+
+1. Test case: `edit-wedding 2 w/John's Wedding a/Address1 d/2024-12-31 `<br>
+   Expected: No weddings are edited. Error details shown.
+
+### Assigning Wedding
+##### ... if `John's Wedding` is in the list && if the first contact is not already in John's Wedding
+1. Test case: `assign-wedding 1 w/John's Wedding`<br>
+   Expected: 1st contact is put in `John's Wedding`'s guest list and 1st contact has `John's Wedding` on its contact card.
+
+1. Test case: `assign-wedding 1 w/John's Wedding p1/`<br>
+   Expected: 1st contact is put as `John's Wedding`'s Partner 1 and 1st contact has `John's Wedding` on its contact card.
+
+1. Test case: `assign-wedding 1 w/John's Wedding p2/`<br>
+   Expected: 1st contact is put as `John's Wedding`'s Partner 2 and 1st contact has `John's Wedding` on its contact card.
+
+1. Test case: `assign-wedding 1 w/John's Wedding p1/ p2/`<br>
+   Expected: 1st contact is put as `John's Wedding`'s Partner 1 and 1st contact has `John's Wedding` on its contact card.
+
+##### ... if `John's Wedding` is in the list && if the first contact is already in John's Wedding
+1. Test case: `assign-wedding 1 w/John's Wedding [p1/] [p2/]`<br>
+   Expected: No weddings are assigned to any contacts. Error details shown.
+
+##### ... if `John's Wedding` is not in the list
+1. Test case: `assign-wedding 1 w/John's Wedding [p1/] [p2/]`<br>
+   Expected: No contact is assigned. Error details shown.
+
+1. Test case: `assign-wedding 1 w/John's Wedding f/`<br>
+   Expected: 1st contact is put in `John's Wedding`'s guest list and 1st contact has `John's Wedding` on its contact card.
+
+### Unassigning Wedding
+##### ... if first contact in `John's Wedding`
+1. Test case: `unassign-wedding 1 w/John's Wedding`<br>
+   Expected: 1st contact is removed from `John's Wedding` and `John's Wedding` is removed from their contact card.
+
+##### ... if first contact not in `John's Wedding`
+1. Test case: `unassign-wedding 1 w/John's Wedding`<br>
+   Expected: No weddings are unassigned. Error details shown.
+
+---
+<h3 class="features">Task Features</h3>
+
+### Creating Task
+##### ... if `Task1` is not in the list of tasks
+1. Test case: `create-task tk/Task1`<br>
+   Expected: `Task1` is added to list of tasks. Details of the added task are shown.
+
+##### ... if `Task1` is in the list of tasks
+1. Test case: `create-task tk/Task1`<br>
+   Expected: No tasks are added to list of tasks. Error details shown.
+
+### Deleting Task
+##### ... if there is only 1 task in the list
+1. Test case: `delete-task 1`<br>
+   Expected: `Task1` is removed from list of tasks. Details of the removed task are shown.
+
+1. Test case: `delete-task 2`<br>
+   Expected: No tasks are removed from list of tasks. Error details shown.
+
+### Assigning Task
+##### ... if there's only 1 task and 1 vendor without that task assigned to them
```

Comment:
I think for these cases, need to make it clear which of 0, 1, 2 are vendors (if all of them, state at the beginning) and also make it clear that it is number 1 without the task assigned (implied but may not be clear)

Reply:
there's only 1 vendor so 0 and 2 are invalid indexes

Reply:
will rephrase